### PR TITLE
fix: add more caching to intervals and cron

### PR DESCRIPTION
**Pull request by tobymao**

### CODE REVIEW

File: sqlmesh/core/plan/definition.py [hunk 1]
```diff
@@ -332,7 +332,10 @@ def has_changes(self) -> bool:
 
     def is_new_snapshot(self, snapshot: Snapshot) -> bool:
         """Returns True if the given snapshot is a new snapshot in this plan."""
-        return snapshot.snapshot_id in {s.snapshot_id for s in self.new_snapshots}
```
Comment (tobymao) [inline sqlmesh/core/plan/definition.py:335]: @eakmanrq you missed this section in your refactor, it's an N^2 operation

Comment (eakmanrq) [reply sqlmesh/core/plan/definition.py:335]: Thanks for catching. Want to just update `new_snapshots` so it is a dict of snapshot ID to snapshot?

Comment (tobymao) [reply sqlmesh/core/plan/definition.py:335]: ok i can do that

**Review by izeigerman — APPROVED** [review]

File: sqlmesh/core/plan/definition.py (hunk 1 above)
Comment (tobymao) [reply sqlmesh/core/plan/definition.py:335]: decided not to do that because then i have to change other places to values

**Pull request merged**